cmd/discovery-server/app/app.go

```diff
@@ -147,6 +147,7 @@ func run(ctx context.Context, log logr.Logger, opts *options.Config) error {
 		"GET "+jwksPath,
 		metrics.InstrumentHandler(jwksPath, http.HandlerFunc(h.HandleJWKS)),
 	)
+	mux.Handle("/", http.HandlerFunc(h.HandleNotFound))
```

Comment:
As discussed offline, let's enhance `HandleWellKnown` and `HandleWellKnown` to disallow methods other than `GET` and `HEAD`.